tests/js/client/aql/vector/aql-vector-agency-wrong-definition-cluster.js

```diff
@@ -0,0 +1,200 @@
+// / Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+// /
+// / Licensed under the Business Source License 1.1 (the "License");
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+/// @author Jure Bajic
+// //////////////////////////////////////////////////////////////////////////////
+
+const internal = require("internal");
+const jsunity = require("jsunity");
+const arangodb = require("@arangodb");
+const helper = require("@arangodb/aql-helper");
+const aql = arangodb.aql;
+const getQueryResults = helper.getQueryResults;
+const assertQueryError = helper.assertQueryError;
+const errors = internal.errors;
+const db = internal.db;
+const {
+    randomNumberGeneratorFloat,
+} = require("@arangodb/testutils/seededRandom");
+const dbName = "vectorDb";
+const collName = "vectorColl";
+const indexName = "vectorIndex";
+let {
+    agency,
+    getMetric,
+    getEndpointById
+} = require('@arangodb/test-helper');
+
+let IM = global.instanceManager;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function VectorIndexCorrectDefinitionInAgencyTest() {
+    let collection;
+    const dimension = 500;
+    const seed = 12132390894;
+
+    return {
+        setUp: function() {
+            db._createDatabase(dbName);
+            db._useDatabase(dbName);
+
+            collection = db._create(collName, {
+                numberOfShards: 3
+            });
+
+            let docs = [];
+            let gen = randomNumberGeneratorFloat(seed);
+            for (let i = 0; i < 100; ++i) {
+                const vector = Array.from({
+                    length: dimension
+                }, () => gen());
+                docs.push({
+                    vector
+                });
+            }
+            collection.insert(docs);
+        },
+
+        tearDown: function() {
+            db._useDatabase("_system");
+            db._dropDatabase(dbName);
+        },
+
+        testVectorIndexCreationCorrect: function() {
+            let beforeVersion = IM.agencyMgr.get("Current/Version").arango.Current.Version;
+            collection.ensureIndex({
+                "name": "test",
+                "params": {
+                    "dimension": dimension,
+                    "metric": "l2",
+                    "nLists": 1,
```

Comment:
there is also the `factory` parameter, maybe you want to write a test for that as well?

Reply:
Ah nice catch, I will add it. Thanks! :smiley: